src/ui/public/filter_bar/filter_bar_click_handler.js

```diff
@@ -2,21 +2,11 @@ define(function (require) {
   var _ = require('lodash');
   var dedupFilters = require('./lib/dedupFilters');
   var uniqFilters = require('./lib/uniqFilters');
-
-  // given an object or array of objects, return the value of the passed param
-  // if the param is missing, return undefined
-  function findByParam(values, param) {
-    if (_.isArray(values)) { // point series chart
-      var index = _.findIndex(values, param);
-      if (index === -1) return;
-      return values[index][param];
-    }
-    return values[param]; // pie chart
-  }
+  var findByParam = require('ui/utils/find_by_param');
 
   return function (Notifier) {
     return function ($state) {
-      return function (event) {
+      return function (event, simulate) {
```

Comment:
Added `simulate` so you can check whether or not a filter would be created by this function. Allows us to check if we should display the filter buttons without reimplementing all of the logic in this file somewhere else.